packages/odf/redux/provider-hooks/useODFNamespace.ts

```diff
@@ -25,11 +26,18 @@ const namespaceDetector = async (maxAttempt = 5): Promise<string> => {
         model: SubscriptionModel,
         queryParams: { ns: null },
       })) as SubscriptionKind[];
-      ns = getNamespace(
-        subscriptions.find((subscription) =>
-          subscription.spec.name.includes(SPEC_NAME)
-        )
+      const odfSubscription = subscriptions.find((subscription) =>
+        getName(subscription).includes(ODF_SUBSCRIPTION_NAME)
       );
+      const isODFPresent = odfSubscription !== undefined;
+      if (isODFPresent) {
+        ns = getNamespace(odfSubscription);
+      } else {
```

Comment:
should we add it ?? just for safety... this will ensure mostly all of existing functionality of this hook will remain intact..
```suggestion
      } else if (!isProviderMode) {
```

Reply:
my bad... my intension was to not run client fallback **at all** if it's normal ODF mode deployment (to make it how existing flow works)... but I guess we can't determine that !!